test/python/transpiler/test_unitary_synthesis.py

```diff
@@ -843,19 +847,110 @@ def test_iswap_no_cx_synthesis_succeeds(self):
         result_qc = dag_to_circuit(result_dag)
         self.assertTrue(np.allclose(Operator(result_qc.to_gate()).to_matrix(), cxmat))
 
-    def test_parameterized_basis_gate_in_target(self):
-        """Test synthesis with parameterized RXX gate."""
+    @combine(is_random=[True, False], param_gate=[RXXGate, RZZGate, CPhaseGate])
+    def test_parameterized_basis_gate_in_target(self, is_random, param_gate):
+        """Test synthesis with parameterized RZZ/RXX gate."""
         theta = Parameter("θ")
         lam = Parameter("λ")
+        phi = Parameter("ϕ")
         target = Target(num_qubits=2)
         target.add_instruction(RZGate(lam))
-        target.add_instruction(RXGate(theta))
-        target.add_instruction(RXXGate(theta))
+        target.add_instruction(RXGate(phi))
+        target.add_instruction(param_gate(theta))
         qc = QuantumCircuit(2)
+        if is_random:
+            qc.unitary(random_unitary(4, seed=1234), [0, 1])
         qc.cp(np.pi / 2, 0, 1)
         qc_transpiled = transpile(qc, target=target, optimization_level=3, seed_transpiler=42)
         opcount = qc_transpiled.count_ops()
-        self.assertTrue(set(opcount).issubset({"rz", "rx", "rxx"}))
+        self.assertTrue(set(opcount).issubset({"rz", "rx", param_gate(theta).name}))
+        self.assertTrue(np.allclose(Operator(qc_transpiled), Operator(qc)))
+
+    def test_custom_parameterized_gate_in_target(self):
+        """Test synthesis with custom parameterized gate in target."""
+
+        class CustomXXGate(RXXGate):
+            """Custom RXXGate subclass that's not a standard gate"""
+
+            _standard_gate = None
+
+            def __init__(self, theta, label=None):
+                super().__init__(theta, label)
+                self.name = "MyCustomXXGate"
+
+        theta = Parameter("θ")
+        lam = Parameter("λ")
+        phi = Parameter("ϕ")
+
+        target = Target(num_qubits=2)
+        target.add_instruction(RZGate(lam))
+        target.add_instruction(RXGate(phi))
+        target.add_instruction(CustomXXGate(theta))
+
+        qc = QuantumCircuit(2)
+        qc.unitary(random_unitary(4, seed=1234), [0, 1])
+        qc_transpiled = UnitarySynthesis(target=target)(qc)
+        # TODO: fix this assertion
+        opcount = qc_transpiled.count_ops()
+        self.assertTrue(set(opcount).issubset({"rz", "rx", "MyCustomXXGate"}))
+        # self.assertTrue(np.allclose(Operator(qc_transpiled), Operator(qc)))
```

Comment:
Note that this test is still failing, since in the decomposed circuit I still see `MyCustomXXGate(theta)` instead of the relevant angles.

Reply:
these is the place to fix:
https://github.com/Qiskit/qiskit/blob/f4da2dac064a0442bf1ed091679e16bc11af3cc8/crates/accelerate/src/unitary_synthesis.rs#L157-L164